addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/discovery/SomfyTahomaItemDiscoveryService.java

```diff
@@ -0,0 +1,277 @@
+                deviceDiscovered(device, THING_TYPE_AWNING);
+                break;
+            case CONTACTSENSOR:
+                deviceDiscovered(device, THING_TYPE_CONTACTSENSOR);
+                break;
+            case EXTERIORSCREEN:
+                deviceDiscovered(device, THING_TYPE_EXTERIORSCREEN);
+                break;
+            case EXTERIORVENETIANBLIND:
+                deviceDiscovered(device, THING_TYPE_EXTERIORVENETIANBLIND);
+                break;
+            case GARAGEDOOR:
+                deviceDiscovered(device, THING_TYPE_GARAGEDOOR);
+                break;
+            case LIGHT:
+                deviceDiscovered(device, THING_TYPE_LIGHT);
+                break;
+            case LIGHTSENSOR:
+                deviceDiscovered(device, THING_TYPE_LIGHTSENSOR);
+                break;
+            case OCCUPANCYSENSOR:
+                deviceDiscovered(device, THING_TYPE_OCCUPANCYSENSOR);
+                break;
+            case ONOFF:
+                deviceDiscovered(device, THING_TYPE_ONOFF);
+                break;
+            case ROLLERSHUTTER:
+                if (isSilentRollerShutter(device)) {
+                    deviceDiscovered(device, THING_TYPE_ROLLERSHUTTER_SILENT);
+                } else {
+                    deviceDiscovered(device, THING_TYPE_ROLLERSHUTTER);
+                }
+                break;
+            case SCREEN:
+                deviceDiscovered(device, THING_TYPE_SCREEN);
+                break;
+            case SMOKESENSOR:
+                deviceDiscovered(device, THING_TYPE_SMOKESENSOR);
+                break;
+            case VENETIANBLIND:
+                deviceDiscovered(device, THING_TYPE_VENETIANBLIND);
+                break;
+            case WINDOW:
+                deviceDiscovered(device, THING_TYPE_WINDOW);
+                break;
+            case ALARM:
+                if (device.getDeviceURL().startsWith("internal:")) {
+                    deviceDiscovered(device, THING_TYPE_INTERNAL_ALARM);
+                } else {
+                    deviceDiscovered(device, THING_TYPE_EXTERNAL_ALARM);
+                }
+                break;
+            case POD:
+                deviceDiscovered(device, THING_TYPE_POD);
+                break;
+            case HEATINGSYSTEM:
+                if (isOnOffHeatingSystem(device)) {
+                    deviceDiscovered(device, THING_TYPE_ONOFF_HEATING_SYSTEM);
+                } else {
+                    deviceDiscovered(device, THING_TYPE_HEATING_SYSTEM);
+                }
+                break;
+            case DOORLOCK:
+                deviceDiscovered(device, THING_TYPE_DOOR_LOCK);
+                break;
+            case PERGOLA:
+                deviceDiscovered(device, THING_TYPE_PERGOLA);
+                break;
+            case PROTOCOLGATEWAY:
+                break;
+            default:
+                logger.warn("Detected a new unsupported device: {}", device.getUiClass());
```

Comment:
Usually, devices that are not known should simply be ignored. A warning therefore seems not suitable here.
If you want to encourage people to report new devices, you might do INFO logging and also add a sentence that asks the user to file it as an issue.

Reply:
Changed to info and asked an user to create a new issue